diag/path.go

```diff
@@ -0,0 +1,16 @@
+package diag
+
+import "github.com/zclconf/go-cty/cty"
+
+func JoinPath(parent cty.Path, child cty.Path) cty.Path {
```

Comment:
Does this need to be public? Also, I wonder if we can get away with putting this sort of pure cty functionality in another (internal?) package, as it isn't `diag`-specific.

Reply:
does not need to be public. It's somewhat niche logic honestly I can just encapsulate it in place where it used I don't suspect it will be used again.